R/yearly_stat.R

```diff
@@ -0,0 +1,279 @@
+# Copyright 2015 Province of British Columbia
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+# Yearly statistics.
+yearly_stat <- function(data, dt = "date", val = "value", 
+                        by = c("ems_id", "site"), 
+                        stat, stat.opts = NULL, quarter_units = "prop",
+                        exclude_df, exclude_df_dt) {
+  # Get quarter validity
+  quarter_valid <- valid_by_quarter(data, dt, by, quarter_units)
+  # Exclude data.
+  if(!is.null(exclude_df)) data <- exclude_df(data, dt, by, exclude_df, exclude_df_dt)
```

Comment:
I'm getting an error when running `o3_annual_4th_highest` with exclusions. I think this should be `exclude_data()`